spec/requests/api/v1/users_spec.rb

```diff
@@ -81,6 +81,37 @@
           expect(parsed_response["data"].size).to eql(0)
         end
       end
+
+      context "when filtering by identity record email" do
+        let(:user) { create(:user, email: "fred@example.com") }
+        let!(:identity) { create(:participant_identity, user:, email: "charlie@example.com") }
+
+        context "when a matching identity record exists" do
```

Comment:
could we add a test case where the user has no identity attached to it and verify that it still returns this user (or maybe it should actually not since this user may not be in a valid state?

you'll find a none negligible number of users without participant_identities:
`User.left_outer_joins(:participant_identities).where(participant_identities: {id: nil}).count`

and also recently created:
`ser.left_outer_joins(:participant_identities).where(participant_identities: {id: nil}).order(created_at: :desc).first.created_at` == `today`

Reply:
What are the consequences of returning such user to NPQ registration service? @Lockyy @peteryates

Reply:
Test case added. tksss

Reply:
I think a `User` can be valid if there are no identities attached because `User` is directly linked to some other kinds of profile (`AppropriateBodyProfile`, `TeacherProfile`). The tests look good like they're providing good coverage though.

Reply:
It would also lead to further issues if we didn't return a user because then we'd be in a similar situation where a user record is blocking record creation but isn't surfaced by the lookup.